src/core/cgroup.c

```diff
@@ -4085,17 +4085,18 @@ int unit_get_memory_accounting(Unit *u, CGroupMemoryAccountingMetric metric, uin
                 return r;
         updated = r >= 0;
 
-cache:
-        if (metric > _CGROUP_MEMORY_ACCOUNTING_METRIC_CACHED_LAST)
-                return -ENODATA;
+finish:
+        if (metric <= _CGROUP_MEMORY_ACCOUNTING_METRIC_CACHED_LAST) {
+                uint64_t *last = &u->memory_accounting_last[metric];
```

Comment:
if you touch this, maybe write this as:

```c
uint64_t last = u->memory_accounting_last + metric;
```

I always find it weird to first deref this and then get a pointer of it. 

but doesn't matter